clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -4431,6 +4431,26 @@ def AssumeSepStorageOp : CIR_Op<"assume.separate_storage", [SameTypeOperands]> {
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// PtrMask Operations
+//===----------------------------------------------------------------------===//
+
+def PtrMaskOp : CIR_Op<"ptr_mask"> {
+  let summary = "Corresponding to LLVM's ptr mask operation";
+  let description = [{
+    The `cir.ptr_mask` operation takes a pointer and an interger `mask` as its
+    argument and return the masked pointer type according to the `mask`.
+  }];
+
+  let arguments = (ins CIR_PointerType:$ptr,
+                       CIR_IntType:$mask);
+  let results = (outs CIR_PointerType:$result);
```

Comment:
The input/output ptrs should be the same type, right? You should add a constraint to that effect too (it will make the printing slightly nicer too).